File: wdio-ios-app/tests/pages/login.page.ts
import { $ } from '@wdio/globals';

class LoginPage {
  get bobUserButton() {
    return $('~bob@example.com');
  }

  get visualUserButton() {
    return $('~visual@example.com');
  }

  get loginButton() {
    return $('~Login');
  }

  /**
   * Prefil with Visual User
   */
  async clickVisualUserButton() {
    await this.visualUserButton.click();
  }

  /**
   * Prefil with Bob User
   */
    async clickBobUserButton() {
      await this.bobUserButton.click();
    }

  /**
   * Do Login
   */
  async clickLogin() {
    await this.loginButton.click();
  }
}

export default new LoginPage();

File: java-ios-app/src/test/java/com/example/pageobjects/LoginPage.java
package com.example.pageobjects;

import io.appium.java_client.AppiumBy;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.remote.RemoteWebDriver;

public class LoginPage {

    private final RemoteWebDriver driver;

    public LoginPage(RemoteWebDriver driver) {
        this.driver = driver;
    }

    public WebElement getBobUserButton() {
        return driver.findElement(AppiumBy.accessibilityId("bob@example.com"));
    }

    public WebElement getVisualUserButton() {
        return driver.findElement(AppiumBy.accessibilityId("visual@example.com"));
    }

    public WebElement getLoginButton() {
        return driver.findElement(AppiumBy.accessibilityId("Login"));
    }

    public void clickBobUserButton() {
        getBobUserButton().click();
    }

    public void clickVisualUserButton() {
        getVisualUserButton().click();
    }

    public void clickLoginButton() {
        getLoginButton().click();
    }

}
